### PR TITLE
Add network config for defaults, add fallbacks to NearProvider constructor, remo…

### PR DESCRIPTION
**Pull request by mikedotexe**
…ve creation of test acct in constructor

As discussed here:
https://github.com/near/balancer-core/pull/1#discussion_r537674354

Instead of having projects specify the `nodeUrl` every time, we can have them give the `networkId` and `masterAccount` (via environment variable) and have the network details living in this repo.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,43 +3,40 @@ const assert = require('bsert');
...
         this.accountEvmAddress = utils.nearAccountToEvmAddress(this.accountId);
         this.accounts = new Map();
         this.accounts.set(this.accountId, this.account);
-
-        if (params.numTestAccounts) {
-            // Creates test accounts if given parameter is passed.
-            utils.createTestAccounts(this.account, params.numTestAccounts)
-                .then(() => {}).catch((error) => { throw error; });
-        }
+        this.walletUrl = walletUrl || process.env.NEAR_WALLET_URL || networkDefaults.walletUrl || `https://wallet.${networkId}.near.org`;
```
Comment (artob) [inline src/index.js:38]: What about http://wallet.mainnet.near.org? That wouldn't work. But OK, I suppose that last default clause will never be actually utilized, for mainnet.

File: src/network-config.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            };
+        case 'betanet':
+            return {
+                nodeUrl: 'https://rpc.betanet.near.org',
+                networkId: 'betanet',
+                evmAccountId: 'evm',
+                walletUrl: 'https://wallet.testnet.near.org',
+                explorerUrl: 'https://explorer.testnet.near.org',
+            };
+        case 'local':
```
Comment (artob) [inline src/network-config.js:27]: Is it always specified everywhere as "local", or can it be sometimes "localnet"? I think I've seen the latter in some docs, but not sure.

File: src/network-config.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                walletUrl: 'https://wallet.testnet.near.org',
+                explorerUrl: 'https://explorer.testnet.near.org',
+            };
+        case 'betanet':
+            return {
+                nodeUrl: 'https://rpc.betanet.near.org',
+                networkId: 'betanet',
+                evmAccountId: 'evm',
+                walletUrl: 'https://wallet.testnet.near.org',
+                explorerUrl: 'https://explorer.testnet.near.org',
```
Comment (artob) [inline src/network-config.js:25]: Looks like `walletUrl` and `explorerUrl` are incorrect, here, for betanet.

**Review by artob — CHANGES_REQUESTED** [review]
LGTM overall, but will need to correct the `walletUrl` and `explorerUrl` links for betanet.

File: src/index.js (hunk 1 above)
Comment (mikedotexe) [reply src/index.js:38]: Yes, we haven't really thought out the naming conventions, I'd say. This isn't a catch-all, but a reasonable, final fallback I believe. And if it fails it's at least something that can be reported and understood more easily. There are previous cases where we've ran into this kind of trouble before. Glad you spotted it. In this case, I think it's fine to keep as is unless others feel differently.

File: src/network-config.js (hunk 2 above)
Comment (mikedotexe) [reply src/network-config.js:27]: Good question. Indeed `nearup` creates files in `~/.near/locanet` which might be where you recognize that.
In terms of configuration, we use 'local' for this.
https://github.com/near/near-cli/blob/91a0cba74eb51709fa18568802ed36bccfaa2dcc/config.js#L38
and then all our examples with config, like for example:
https://github.com/near-examples/guest-book/blob/b4557f0b5eecffe522c0acc3ce75ac325a3ee2bb/src/config.js#L33

File: src/network-config.js (hunk 3 above)
Comment (mikedotexe) [reply src/network-config.js]: Good eye! pushed the fix. thank you

**Review by artob — APPROVED** [review]
LGTM :+1: Merging...

**Pull request merged**